### DISCUSSION TITLE
How can I 'close' a polyline with the new last vertex and marker at the same position at the start ones?

### DISCUSSION THREAD

**Discussion by volozhanin-gg** [geoman-io/leaflet-geoman#985] [Questions & Help]
The 'closed' polyline is for the polygonal selection tool.
Unlike the polygon, the polyline will not take mouse events. 
Well, the real closed polyline would be the best,
but something prevents you from implementing it.

**Comment by Falke-Design** [comment]
I don't really understand your needs ... If you want a polygon (same start and end point) without a fill, you can disable it in the [style](https://leafletjs.com/reference-1.7.1.html#path):

```
map.pm.enableDraw('Polygon',{pathOptions: {fill: false}})
```

Can you please share a example what do you mean with `will not take mouse events`

**Reply by volozhanin-gg** [reply]
I want to make a polygonal selection tool. 

I have a lot of intersecting polygons of very different sizes. Some of them are much bigger than the selection area, so I get only polygons containing, not intersecting the selection polygon.

I already implemented separate polygons' selection by mouse clicks. I also implemented multiple polygons' selection by adding edited polygons, with the data polygons' containment calculation in PostGIS. It's ok (drag, cut, markers edit for the selection polygons), because the added polygons are a bit higher because of their presence in the end of the layers' list.

![image](https://user-images.githubusercontent.com/90439312/132821360-6789d69d-3199-4a4b-891a-e00372f87e73.png)

The data polygons are red, the selected polygons are blue, the selection polygons (with transparent area's color) are green.

But blue polygons became unclickable, because they are now behind the selection polygons. The user still should be able to unselect some of them. I cannot get the data polygons on the pane higher than selection polygons because they can be big.

Closed polyline will NOT take mouse events in it's area, unlike the polygon.

I cannot make   green_selection_layer.interactive = false
because it cannot be changed or dragged when it's created

May be I can make the selection polygon's inside area transparent for all mouse events to let them come to the selected polygons and other objects inside?

I cannot replace my data polygons with their polygonal areas because they have to be filled with color and should be selected by clicks in their areas.

We have a snippet: https://stackoverflow.com/questions/42092095/how-to-complete-a-polyline-in-leaflet-draw-after-clicking-second-point

but 1) I was not find the function like this in geoman 2) I would like to do it outside the geoman's code, may be in map.on() and layer.on()

thanks anyway

**Reply by Falke-Design** [reply]
Ok I undestand. I suggest you to replace the selection polygon with a polyline after drawing:
```
      var selection;
      map.on('pm:create',(e)=>{
        var options = e.layer.options;
        options.fill = false;

        var latlngs = e.layer.getLatLngs()[0]; // get first array because polygons are not flat
        latlngs.push(latlngs[0]); // Add first latlng to the end -> closed polyline
        selection = L.polyline(latlngs, options ).addTo(map);
        e.layer.removeFrom(map);
      });
```

**Reply by volozhanin-gg** [reply]
layer._path.style.pointerEvents = 'stroke'
makes the polygon's svg item interrior transparent for mouse events
(so layer can be dragged and removed only by border)

but I worry I did not find an event when the svg item is recreated to set the style property again
if it's not the point, we have done it
